components/dot_menu/dot_menu.jsx

```diff
@@ -218,16 +218,19 @@ export default class DotMenu extends React.PureComponent {
         </Tooltip>
     )
 
-    refCallback = (ref) => {
-        if (ref) {
-            const rect = ref.rect();
-            const y = rect.y || rect.top;
+    refCallback = (menuRef) => {
+        if (menuRef) {
+            const rect = menuRef.rect();
+            const buttonRef = this.buttonRef.current.getBoundingClientRect();
+            const y = buttonRef.y || buttonRef.top;
```

Comment:
while it shouldn't be 0 ever, this would be a little safer in case something goes wrong, it is also more clear on why are we checking both values.
```suggestion
            const y = typeof buttonRef.y !== 'undefined' ? buttonRef.y ? buttonRef.top;
```